core/src/libraries/constants/Constants.sol

```diff
@@ -29,6 +29,20 @@ uint256 constant IV_SCALE = 24 hours;
 
 uint256 constant IV_CHANGE_PER_SECOND = 5e12;
 
-uint256 constant FEE_GROWTH_GLOBALS_SAMPLE_PERIOD = 1 minutes;
-
-uint32 constant ORACLE_LOOKBACK = 20 minutes;
+/// @dev To estimate volume, we need 2 samples. One is always at the current block, the other is from
+/// `FEE_GROWTH_AVG_WINDOW` seconds ago, +/- `3 * FEE_GROWTH_SAMPLE_PERIOD`. Larger values make the resulting volume
+/// estimate more robust, but may cause the oracle to miss brief spikes in activity.
+uint256 constant FEE_GROWTH_AVG_WINDOW = 6 hours;
+
+/// @dev The length of the circular buffer that stores feeGrowthGlobals samples.
+/// Must have be in interval [ FEE_GROWTH_AVG_WINDOW / FEE_GROWTH_SAMPLE_PERIOD, 256 )
+uint256 constant FEE_GROWTH_ARRAY_LENGTH = 72;
+
+/// @dev The minimum number of seconds that must elapse before a new feeGrowthGlobals sample will be stored. This also
+/// controls how often the oracle can update IV.
+uint256 constant FEE_GROWTH_SAMPLE_PERIOD = 5 minutes;
+
+/// @dev To compute Uniswap mean price & liquidity, we need 2 samples. One is always at the current block, the other is
+/// from `UNISWAP_AVG_WINDOW` seconds ago. Larger values make the resulting price/liquidity values harder to
+/// manipulate, but also make the oracle slower to respond to changes.
+uint32 constant UNISWAP_AVG_WINDOW = 30 minutes;
```

Comment:
I'm curious, how did you come up with these? I'm assuming the effect increasing the `FEE_GROWTH_SAMPLE_PERIOD` from 1 minute to 5 minutes as well as the `UNISWAP_AVG_WINDOW` from 20 minutes to 30 minutes make it more robust against spikes in price, but I'm wondering why these values in particular?

Reply:
I changed `UNISWAP_AVG_WINDOW` to 30 minutes based on Euler's research. I increased `FEE_GROWTH_SAMPLE_PERIOD` to 5 minutes so that we could keep the ring buffer a reasonable length while extending the `FEE_GROWTH_AVG_WINDOW` to 6 hours. I wanted 6 hours on that one to reduce the likelihood of sampling over an hour in which there was no trading activity.